syncer/syncer.go

```diff
@@ -42,17 +44,28 @@ type PeerInfo struct {
 
 // A PeerStore stores peers and bans.
 type PeerStore interface {
-	AddPeer(peer string)
-	Peers() []string
-	UpdatePeerInfo(peer string, fn func(*PeerInfo))
-	PeerInfo(peer string) (PeerInfo, bool)
-
+	// AddPeer adds a peer to the store. If the peer already exists, nil should
+	// be returned.
+	AddPeer(peer string) error
+	// Peers returns the set of known peers.
+	Peers() ([]PeerInfo, error)
+	// UpdatePeerInfo updates the metadata for the specified peer. If the peer
+	// is not found, the error should be ErrPeerNotFound.
+	UpdatePeerInfo(peer string, fn func(*PeerInfo)) error
 	// Ban temporarily bans one or more IPs. The addr should either be a single
 	// IP with port (e.g. 1.2.3.4:5678) or a CIDR subnet (e.g. 1.2.3.4/16).
-	Ban(addr string, duration time.Duration, reason string)
-	Banned(peer string) bool
+	Ban(addr string, duration time.Duration, reason string) error
+
+	// Banned returns nil if the peer is not banned. Otherwise, ErrPeerBanned
+	// is returned.
+	Banned(peer string) error
```

Comment:
sentinel values are tempting, but I prefer `(bool, error)` for this sort of thing. Compare:
```go
if banned, err := ps.Banned(peer); err != nil {
    // handle err
} else if banned {
    // handle banned
}
```
vs.
```go
if err := ps.Banned(peer); err != nil && err != ErrPeerBanned {
    // handle err
} else if err == ErrPeerBanned {
    // handle banned
}

// or: 

if err := ps.Banned(peer); err == ErrPeerBanned {
    // handle banned
} else if err != nil {
    // handle err
}

// or:

if ps.Banned(peer) == ErrPeerBanned {
    // handle banned
} else if ps.Banned(peer) != nil {
    // handle err
}
```
all of these are dissatisfactory imo.